build/Cache.js

```diff
@@ -0,0 +1,228 @@
+
+import { Node } from './Node';
+import { PriorityQueue } from './Queue';
+
+const RNFetchBlob = require("react-native-fetch-blob").default;
+const SHA1 = require("crypto-js/sha1");
+const s4 = () => Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
+const BASE_DIR = RNFetchBlob.fs.dirs.DocumentDir + "/imageFiles";
+const QUEUE_DIR = RNFetchBlob.fs.dirs.DocumentDir + "/";
+const CACHE_LIMIT_FILE_COUNT = 10;
+
+export class ImageCache {
+
+    static readInQueue(file) {
+      return new Promise((resolve, reject) => {
+        RNFetchBlob.fs.readFile(QUEUE_DIR + file, 'utf8')
+        .then((data) => resolve(JSON.parse(data)))
+        .catch((err) => {
+          console.log('No Queue File Found: ', err);
+          RNFetchBlob.fs.exists(QUEUE_DIR + file).then((exists) => {
+             if (exists) {
+               RNFetchBlob.fs.writeFile(QUEUE_DIR + file, '[]', 'utf8')
+               .then((result) => resolve([]))
+               .catch((err) => reject(err))
+             }
+             else {
+               RNFetchBlob.fs.createFile(QUEUE_DIR + file, '[]', 'utf8')
+               .then((result) => resolve([]))
+               .catch((err) => reject(err))
+             }
+         });
+        });
+      })
+    }
+
+    static save(file, queueData) {
+      return new Promise((resolve, reject) => {
+        const data = JSON.stringify(queueData)
+        RNFetchBlob.fs.writeFile(QUEUE_DIR + file, data, 'utf8')
+        .then((result) => (process.env.NODE_ENV === "TEST") ? console.log("") : console.log('Queue successfully saved!') )
+        .catch((err) => reject(err))
+      })
+    }
+
+    prepareMem(data){
+      let queue = new PriorityQueue();
+      const map = {};
+      data.forEach((d) => {
+        let n = new Node(d)
+        queue.enqueue(n)
+        map[d] = n;
+      })
+      return {queue, map}
+    }
+
+    constructor(data) {
+        this.cache = {};
+        const { queue, map } = this.prepareMem(data)
+        this.queue = queue;
```

Comment:
let's try to split this Cache into two objects the LRUCache and the FirebaseImageCache to separate the LRU logic from the reading/writing to file  & download from firebase.